### ISSUE TITLE
command line '-install' option does not work anymore (neither does any of the other options)

### ISSUE THREAD

**Issue by vtacquet** [Ylianst/MeshCentral#4565]
**Describe the bug**
./meshagent -install is no longer working (and -update and -connect neither)

**Screenshots**
![CleanShot 2022-09-20 at 15 22 04](https://user-images.githubusercontent.com/5128967/191269124-573f0f44-d90e-47a7-aabf-f93bd2a13fec.png)

**Server Software (please complete the following information):**
 - OS: Ubuntu 20.04
 - Virtualization: Docker 
 - Version: 1.0.85

This happened just after updating to 1.0.85, I was previously on 1.0.64 (so somewhere in between it must have stopped working)

**Comment by dinger1986** [comment]
its been ./meshagent -fullinstall for as long as I can remember and just tested on ubuntu and works fine :)

**Comment by vtacquet** [comment]
no it is not, we use this weekly, it is in a script and -fullinstall also does not work 
![CleanShot 2022-09-20 at 15 47 14](https://user-images.githubusercontent.com/5128967/191274984-dcec9882-f0da-44f6-9bd2-3c73c248a752.png)

**Comment by dinger1986** [comment]
Sorry I thought it was the mesh linux executable.

So what script is that?

**Comment by vtacquet** [comment]
![CleanShot 2022-09-20 at 15 53 21](https://user-images.githubusercontent.com/5128967/191276442-4182dc2a-f241-47a7-95e7-a917f1125277.png)

this one

**Comment by dinger1986** [comment]
are you running it from terminal with a gui? Or CLI?

**Comment by vtacquet** [comment]
scripted/automated in a terminal

**Comment by dinger1986** [comment]
presumably you read this? 
![image](https://user-images.githubusercontent.com/7244447/191281518-80329a19-8133-497d-a247-1a9c0445fcdf.png)

**Comment by vtacquet** [comment]
ofcourse, we install hosts every week, if it was not executable it would not give output 

![CleanShot 2022-09-20 at 16 19 43](https://user-images.githubusercontent.com/5128967/191282957-bcd9f0a4-0571-4c46-a4b1-845743030a0c.png)
![CleanShot 2022-09-20 at 16 20 24](https://user-images.githubusercontent.com/5128967/191283109-026981d5-651d-4c5b-981e-39de7173c1aa.png)

**Comment by dinger1986** [comment]
sorry I meant the gui part, but checking something, wont be long

**Comment by vtacquet** [comment]
it worked until this version, so something has changed

**Comment by dinger1986** [comment]
I think its a new mesh core in it cant remember what version but pretty sure its between your old version and current

**Comment by vtacquet** [comment]
yes but it also does not work without any parameters, so the command in the printscreen above fails... even with a new core it should do what it shows 
![CleanShot 2022-09-20 at 16 24 35](https://user-images.githubusercontent.com/5128967/191284144-27f211f5-4b4e-4269-9093-222760451503.png)

**Comment by si458** [comment]
have you tried using the non binary version? the 'Linix / BSD' operating system

**Comment by dinger1986** [comment]
> yes but it also does not work without any parameters, so the command in the printscreen above fails... even with a new core it should do what it shows

agreed, but its does say about using a gui

Anyways try this (just tested and working on mine)

```
wget -O meshagent 'mymeshurl'
chmod +x meshagnet
env XAUTHORITY=foo DISPLAY=bar ./meshagent -install
```

**Comment by vtacquet** [comment]
![CleanShot 2022-09-20 at 16 27 56](https://user-images.githubusercontent.com/5128967/191285235-187efb83-8bfd-4468-a5eb-2d2146664cca.png)

yes I can still install it manually with the add agent, but that is not the point, we have this automated and now we need to install everything manually

**Comment by vtacquet** [comment]
> > yes but it also does not work without any parameters, so the command in the printscreen above fails... even with a new core it should do what it shows
> 
> agreed, but its does say about using a gui
> 
> Anyways try this (just tested and working on mine)
> 
> ```
> wget -O meshagent 'mymeshurl'
> chmod +x meshagent
> env XAUTHORITY=foo DISPLAY=bar ./meshagent -install
> ```

sure but it changed, that was my point

**Comment by vtacquet** [comment]
and the options stated for "text" don't work

**Comment by si458** [comment]
> we have this automated and now we need to install everything manually

you should be able to reuse the long install command on the 'non binary' option over and over again, its not a 1 time url i believe

**Comment by dinger1986** [comment]
> sure but it changed, that was my point

Yeah but it does say about using with a GUI so its not a bug, maybe that wasnt a requirement before. With adding in the extra `env` prefixing the command it will work fine with your script which was what I was trying to help with :)

**Comment by vtacquet** [comment]
problem is we have the links for the binary for all platforms

**Comment by dinger1986** [comment]
can you share your install script for the binary for all platforms?

**Comment by vtacquet** [comment]
they are all "short" linked and redirected on a traefik proxy and easy to remember if you need to do one manually

`
(rm -f /tmp/mesh; wget --no-check-certificate https://.../mesh/lin -O /tmp/mesh; chmod +x /tmp/mesh; /tmp/mesh -install) 

(rm -f /tmp/mesh; wget --no-check-certificate https://.../mesh/pi -O /tmp/mesh; chmod +x /tmp/mesh; /tmp/mesh -install) 

(rm -f /tmp/mesh; wget --no-check-certificate https://.../mesh/pi64 -O /tmp/mesh; chmod +x /tmp/mesh; /tmp/mesh -install) 

(rm -f /tmp/mesh; wget --no-check-certificate https://.../mesh/mac -O /tmp/mesh; chmod +x /tmp/mesh; /tmp/mesh -install) `

**Comment by dinger1986** [comment]
so can you not change to (dont have a mac so cant test it on that):

```
(rm -f /tmp/mesh; wget --no-check-certificate https://.../mesh/lin -O /tmp/mesh; cd /tmp; chmod +x mesh; env XAUTHORITY=blah DISPLAY=blah ./mesh -install)

(rm -f /tmp/mesh; wget --no-check-certificate https://.../mesh/pi -O /tmp/mesh; cd /tmp; chmod +x mesh; env XAUTHORITY=blah DISPLAY=blah ./mesh -install)

(rm -f /tmp/mesh; wget --no-check-certificate https://.../mesh/pi64 -O /tmp/mesh; cd /tmp; chmod +x mesh; env XAUTHORITY=blah DISPLAY=blah ./mesh -install)

(rm -f /tmp/mesh; wget --no-check-certificate https://.../mesh/mac -O /tmp/mesh; cd /tmp; chmod +x mesh; env XAUTHORITY=blah DISPLAY=blah ./mesh -install)
```

**Comment by si458** [comment]
or generate the command url inside meshcentral and then copy and paste them place of your scripts?
```
(wget "https://meshcentral.myserver.com/meshagents?script=1" -O ./meshinstall.sh || wget "https://meshcentral.myserver.com/meshagents?script=1" --no-proxy -O ./meshinstall.sh) && chmod 755 ./meshinstall.sh && sudo -E ./meshinstall.sh https://meshcentral.myserver.com 'a60HfA@V47SwSlyq5uk8GsAS2j8wvNZZULuyouyNrh4ZdQQ6mDE5N$JlJhEkTQjw' || ./meshinstall.sh https://meshcentral.myserver.com 'a60HfA@V47SwSlyq5uk8GsAS2j8wvNZZULuyouyNrh4ZdQQ6mDE5N$JlJhEkTQjw' 
```

**Comment by vtacquet** [comment]
not all binary installs have a GUI available (was not needed before), and for the generate command -> you can't easily run this manually if you want, you need the entire "ID" which you did not need for the -install parameter

**Comment by vtacquet** [comment]
and still this does not work in "text"
![CleanShot 2022-09-20 at 16 50 21](https://user-images.githubusercontent.com/5128967/191290778-4c5429cd-ac5e-44d5-a812-21b0614fa2c1.png)

**Comment by si458** [comment]
@vtacquet sadly your going to have to either wait until @Ylianst is back off holiday in 3ish weeks time 
OR
use the normal non binary way of installing sadly

**Comment by vtacquet** [comment]
I know, we can continue with an alternative solution for now, but still it is not logic that it gives errors in text, that's why I created the issue

**Comment by si458** [comment]
> I know, we can continue with an alternative solution for now, but still it is not logic that it gives errors in text, that's why I created the issue

you could ask @krayon007 over in the meshagent repo? https://github.com/ylianst/meshagent

**Comment by dinger1986** [comment]
> not all binary installs have a GUI available (was not needed before)

Yes I referenced this above, it wasnt needed before but now it clearly is (for whatever reason) and even states that it is needed, but get what you mean maybe in the text it should point that out or allow it to be installed again in the cli without extra variables needing set. 

Can I suggest changing the name of the issue and add 'Using CLI without GUI'

**Comment by krayon007** [comment]
On a side note, if it fails when you pass in no parameters, (ie, you want the GUI to show up), and nothing happens, you may need to preserve the environment when doing su, such as with `su -` Previously, it would try to find the session id of the logged in user, but this had bad side effects if you were logged in from a non-gui client, or if there were multiple logged in users, etc, which is why it was switched to just use the env variable.
